discovery/pkg/apigee/portalapihandlerjob.go

```diff
@@ -135,18 +136,21 @@ func (j *newPortalAPIHandler) buildServiceBody(newAPI *apigee.APIDocData, produc
 	// create the service body to use for update or create
 	apiID := fmt.Sprint(newAPI.ID)
 
-	// create attributes to be added to revision and instance
+	// create attributes for service agent details
 	serviceDetails := make(map[string]interface{})
+
+	// create attributes to be added to revision and instance
+	serviceAttributes := make(map[string]string)
 	revisionAttributes := make(map[string]string)
 	for k, v := range productAttributes {
-		serviceDetails[k] = v
+		serviceAttributes[k] = v
 		revisionAttributes[k] = v
 	}
 
 	serviceDetails[apigee.ApigeeAgentAttribute.Name] = apigee.ApigeeAgentAttribute.Value
 
 	revisionAttributes[catalogIDKey] = apiID
-	revisionAttributes["PortalID"] = newAPI.PortalID
+	revisionAttributes[portalIDKey] = newAPI.PortalID
```

Comment:
I think PortalID needs to be on x-agent-details right? It is specificed in the MatchAttr func below, but left on the revisionAttributes here.